### PR TITLE
status module (source + tests)

### PR DESCRIPTION
**Pull request by SatyendraBanjare**
added bindings & install instructions for status module functions.

### CODE REVIEW

**Review by Etiene — COMMENTED** [review]
Where are the tests for this module?

**Comment by SatyendraBanjare** [comment]
@Etiene mam, there was an error I was facing earlier so I skipped writing tests for this module. I have added tests now. 

I wanted to ask if I should write bindings for worktree module too...? because I understood while writing tests that file paths(used in one function in test) for files can only be found if they are added to worktree too. 

I also tried after writing index to repo, still the same error happens.

**Comment by Etiene** [comment]
What is the error that you are encountering?

**Comment by Etiene** [comment]
could you solve the conflict too? :)

File: tests/status/test_status_spec.lua [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        -- The file paths for status entries can be found 
+        -- only if they are changed in the worktree as well.
+        --
+        -- Since the index changes DO NOT produce changes in Worktree
+        -- too, therefore if we try to get file paths for
+        -- entry two, it gices error.
+        --
+        --
+        -- Get file paths for entry one.
+        local path_one_old, path_one_new = luagit2.status_file_paths(entry_one)
```
Comment (SatyendraBanjare) [inline tests/status/test_status_spec.lua]: @Etiene  mam, for the latest error, somewhere here. If I try to get the paths for entry two, I get a segfault.

Comment (SatyendraBanjare) [reply tests/status/test_status_spec.lua]: I wrote my understanding for this behaviour  as a note

**Comment by SatyendraBanjare** [comment]
I will resolve the conflicts.

**Comment by SatyendraBanjare** [comment]
@Etiene mam, I fixed the error I was facing. I didn't notice earlier that we need to use `head_to_index->old_file.path` and `head_to_index->new_file.path` to track file path changes in index.

**Comment by Etiene** [comment]
there is still a conflict in the PR though

**Comment by SatyendraBanjare** [comment]
@Etiene  mam, I have done the requested changes please have a look

**Review by Etiene — APPROVED** [review]
👍

**Pull request merged**